adi/ad4020.py

```diff
@@ -45,6 +55,38 @@ def sampling_frequency(self, value):
         """Set the sampling frequency."""
         self._set_iio_dev_attr("sampling_frequency", str(value))
 
+    class _channel_adc(attribute):
+        """AD4000 series differential input voltage channel"""
+
+        # AD4000 series ADC channel
+        def __init__(self, ctrl, channel_name, output):
+            self.name = channel_name
+            self._ctrl = ctrl
+            self._output = output
+
+        @property
+        def raw(self):
+            return self._get_iio_attr(self.name, "raw", self._output)
+
+        @property
+        def scale(self):
+            return float(self._get_iio_attr_str(self.name, "scale", self._output))
+
+        @scale.setter
+        def scale(self, value):
+            self._set_iio_attr(self.name, "scale", False, str(Decimal(value).real))
+
+        @property
+        def scale_available(self):
+            """Provides all available scale(gain) settings for the ADC channel"""
+            return self._get_iio_attr(self.name, "scale_available", False)
+
+        def __call__(self, mV=None):
+            """Convenience function, set / get voltages in SI units (millivolts)"""
+            if mV is not None:
+                self.raw = int(float(mV) / float(self.scale))
+            return self.raw * self.scale
```

Comment:
@machschmitt raw is a read-only attribute, attempting to set throws an error. (the mV=None argument is not needed.) Looks like this was carried over from a DAC device.
Also the formula should include the offset:
((self.raw + self.offset) * self.scale)
so an offset property should also be added. (The device does have an offset attribute.)

Reply:
Yes, that's true. I don't even recall where I got that from but am fixing it now.
The driver doesn't provide offset attribute right now but it may in the future for scenarios where differential ADCs get set in single-ended to differential configuration.